src/main/java/microsoft/exchange/webservices/data/MessageBody.java

```diff
@@ -100,7 +100,8 @@ protected void readAttributesFromXml(EwsServiceXmlReader reader)
   @Override
   protected void readTextValueFromXml(EwsServiceXmlReader reader)
       throws XMLStreamException, ServiceXmlDeserializationException {
-    this.text = reader.readValue();
+    boolean keepWhiteSpace = this.getBodyType() == BodyType.Text;
```

Comment:
no need to store this variable. put it directly into the method below. Also, it is safer to use equals instead of ==